django/utils/text.py

```diff
@@ -101,13 +198,17 @@ def chars(self, num, truncate=None, html=False):
 
         # Calculate the length to truncate to (max length - end_text length)
         truncate_len = length
-        for char in self.add_truncation_text("", truncate):
+        truncate = add_truncation_text("", truncate)
+        for char in truncate:
```

Comment:
I think that this change ought to be reverted. It prematurely resolves the `%(truncated_text)s` placeholder by overriding the `truncate` output. This was probably again related to the zero length issue above.